### ISSUE TITLE
Push Rest API docs do not match Swagger file

### ISSUE THREAD

**Issue by willsam100** [Microsoft/appcenter#177]
**What App Center service does this affect?**
Push notifications with Swagger file

**Describe the bug**
https://docs.microsoft.com/en-us/appcenter/push/rest-api
describes sending a message to a target (eg device) with this type:

```{
   // .. omited for simplicity
    "notification_target" : {
    "type" : "devices_target",
    "devices" : ["00000000-0000-0000-0000-000000000001","00000000-0000-0000-0000-000000000002", "00000000-0000-0000-0000-000000000003"]
  }
}
```

while the type defined in the Swager documentation only contains the `types` field. The extra fiedl (`devices`) is missing from the swagger documentation. 
https://openapi.appcenter.ms/#/push/Push_Send

**To Reproduce**
Generate a client a from the Swagger file - https://openapi.appcenter.ms/
Try to send a push notifiction to a specific device

**Expected behavior**
To be able to generate a client from the swagger file, and send a push notification to a single device. As per the docs. 
The docs to match to the API. In this case, it looks like the swagger client is wrong. 

**Screenshots**
<img width="927" alt="Screen Shot 2019-03-14 at 7 03 02 AM" src="https://user-images.githubusercontent.com/732506/54303255-7315fb00-4627-11e9-8839-e8e217df3fea.png">

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**

**Comment by jwargo** [comment]
@willsam100 thanks for pointing this out. We're aware that the swagger UI doesn't really work very well for the notification types. The team is working on a refresh of this right now (because of some new APIs we're adding to support [Push to User](https://github.com/Microsoft/appcenter/issues/15).  I think the next update will fix this, but I'll make sure.

**Comment by jwargo** [comment]
@willsam100 the swagger contains the reference to the devices array: 

```
NotificationTargetDevices:
    description: Notification target devices. If null, all devices will be targeted(broadcast).
    x-ms-discriminator-value: devices_target
    allOf:
      - $ref: '#/definitions/NotificationTarget'
    properties:
      devices:
        description: List of target devices.
        type: array
        items:
          description: Device Id
          type: string
          format: uuid
    required:
      - devices

```
so I'm not sure why it's not generating the client correctly. When was the last time you tried this? We updated the swagger file within the last week.

**Comment by willsam100** [comment]
@jwargo thanks for this. 

I tried this 24 hours ago. 

It seems that I was not aware `NotificationTargetDevices` existed. The main API (the screenshot I added above) only talks about a `NotificationTarget`. It appears that OpenAI supports a form of inheritance here, and that allows this to work. 

I'm trying this out now (using `NotificationTargetDevices` and NotificationTargetAudiences` ) to see if my generated client works. I assume it will. 

My suggestion would be up docs on the API highlighting that there are other types that can be used. Since using `NotificationTarget` would only be useful when sending to all devices.
